### ISSUE TITLE
Override Javascript hightlights for JSX ?

### ISSUE THREAD

**Issue by tobiwild** [nvim-treesitter/nvim-treesitter#745]
![Screenshot 2020-12-07 at 18 09 38](https://user-images.githubusercontent.com/891533/101382166-ed84b500-38b7-11eb-9cca-771ee4be86db.png)

I like to have all JSX tags in the same color (orange). The upper case tags are mapped as `@constructor` which makes them blue (and I want constructors to be blue). It seems I cannot override this as `javascript/highlights.scm` inherits `jsx/jhighlights.scm` so no matter what I change in `.vim/after/queries/jsx`, this always wins:

```
((identifier) @constructor
 (#match? @constructor "^[A-Z]"))
```

How can I accomplish my goal ?

**Comment by steelsojka** [comment]
This actually seems backwards to me... JSX should be inheriting javascript since JSX is an extensions of javascript. We have some slight rework for javascript/jsx coming up that we can address this in.

**Comment by AlanWarren** [comment]
Thanks for raising this.  I also ran into this issue while trying to override JSX tag attributes.  They are currently using (property_identifier) defined in queries/javascript/highlights.scm.  

I'm new to this, so i figured if I made a more specific query it would win.  (placed in after/queries/jsx/highlights.scm) 
`(jsx_self_closing_element
 attribute: (jsx_attribute (property_identifier)) @character)
`

I was able to get it working, but I had to add the override to the _javascript_ query, which in turn breaks highlighting for tsx files.   Good to hear it's being addressed.
